src/pal/src/misc/dactableaddress.cpp

```diff
@@ -0,0 +1,156 @@
+//
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information. 
+//
+
+/*++
+
+
+
+Module Name:
+
+    dactableaddress.cpp
+
+Abstract:
+
+    Functions to write and read DAC global pointer table address between the coreclr
+    and DAC/debugger processes.
+
+Revision History:
+
+
+--*/
+
+#include "pal/palinternal.h"
+#include <sys/types.h>
+#include <unistd.h>
+#include <stdio.h>
+
+/*++
+Function
+    PAL_PublishDacTableAddress
+
+Parameters
+
+    address
+       [in] address of dac table
+    size
+       [in] size of dac table
+
+Return Values
+       pal errors
+
+--*/
+PALIMPORT
+DWORD
+PALAPI
+PAL_PublishDacTableAddress(
+    IN PVOID address,
+    IN ULONG size)
+{
+    DWORD ret = NO_ERROR;
+
+    // TODO - 3/5/15 - the DAC side needs the debuggee pid
+    // pid_t pid = getpid();
+    pid_t pid = 0;
+    char fileName[100]; 
+    snprintf(fileName, sizeof(fileName), "/tmp/%d_dacTable", pid);
```

Comment:
I think this way is simpler and doesn’t involve separate utilities or scripts that have to know the module format with very little runtime and code size cost.

mikem

From: Jan Kotas [mailto:notifications@github.com]
Sent: Thursday, March 5, 2015 11:12 AM
To: dotnet/coreclr
Cc: Mike McLaughlin
Subject: Re: [coreclr] Implement the DAC global pointer table for Linux (#400)


In src/pal/src/misc/dactableaddress.cpp<https://github.com/dotnet/coreclr/pull/400#discussion_r25891881>:

> +

> +--*/

> +PALIMPORT

> +DWORD

> +PALAPI

> +PAL_PublishDacTableAddress(

> +    IN PVOID address,

> +    IN ULONG size)

> +{

> +    DWORD ret = NO_ERROR;

> +

> +    // TODO - 3/5/15 - the DAC side needs the debuggee pid

> +    // pid_t pid = getpid();

> +    pid_t pid = 0;

> +    char fileName[100];

> +    snprintf(fileName, sizeof(fileName), "/tmp/%d_dacTable", pid);

I would be interested to know why you chose this design, instead of generating the table at build time - how it is done on Windows, and how it used to be done for Silverlight.

—
Reply to this email directly or view it on GitHub<https://github.com/dotnet/coreclr/pull/400/files#r25891881>.